Sources/VaporJWT/AuthError.swift

```diff
@@ -0,0 +1,89 @@
+/// Errors thrown during the parsing, creation
+/// and verification of JWT tokens by this package.
+public struct AuthError: Swift.Error {
+    public let code: Int
+    public let reasonPhrase: String
+    public let origin: Swift.Error?
+
+    init(code: Int, reasonPhrase: String, origin: Swift.Error?) {
+        self.code = code
+        self.reasonPhrase = reasonPhrase
+        self.origin = origin
+    }
+
+    public static func noAuthorizationHeader(_ origin: Swift.Error? = nil) -> Error {
```

Comment:
If you have any other ideas for error patterns you want to try, now is the time to do it. Especially in this package. I'm not totally psyched with what's here as of now, but it solves all of the problems.

Really I'm just trying to recreate an `open enum` as best I can. Maybe we just continue using enums and add an `other()` case that acts as a crutch until ABI stability?